src/classes/UTIL_HtmlOutput_TEST.cls

```diff
@@ -122,4 +122,20 @@ public with sharing class UTIL_HtmlOutput_TEST {
         System.assert(actualHtml.startsWith('&lt;'));
         System.assertNotEquals(html, actualHtml);
     }
+
+    /*******************************************************************************************************
+    * @description Verifies that non-whitelisted javascript event code is stripped from url
+    */
+    @isTest
+    private static void shouldStripJavaScriptFromUrl() {
+        String html = '<a href="https://powerofus.force.com/NPSP_Documentation" onanyjavascriptevent="alert(\'gotcha\')">';
+        string cleanHtml = '<a href="https://powerofus.force.com/NPSP_Documentation" >';
+        UTIL_HtmlOutput_CTRL controller = new UTIL_HtmlOutput_CTRL();
+
+        controller.unsafeHtml = html;
+        controller.hasUrl = true;
+
+        System.assertEquals(cleanHtml, controller.getSafeHtml());
```

Comment:
Would this also work if the bad html is one of following:

- `'<a href=\'https://powerofus.force.com/NPSP_Documentation\' onanyjavascriptevent=\'alert("gotcha")\'>';`
- `'<a href="javascript:alert(\'gotcha\')">'`

Reply:
This is handled by the existing HtmlOutput code.  It essentially just becomes a text string that is never invoked.

Reply:
The first example is stripped by the new code.

Reply:
Confirmed. Thanks!!